lib/bugsnag/integrations/mongo.rb

```diff
@@ -65,6 +65,23 @@ def leave_mongo_breadcrumb(event_name, event)
       Bugsnag.leave_breadcrumb(message, meta_data, Bugsnag::Breadcrumbs::PROCESS_BREADCRUMB_TYPE, :auto)
     end
 
+    ##
+    # Removes values from filter hashes, replacing them with '?'
+    #
+    # @param filter_hash [Hash] the filter hash for the mongo transaction
+    #
+    # @return [Hash] the filtered hash
+    def sanitize_filter(filter_hash)
```

Comment:
This method fails on queries such as `SomeModel.where(tags: ['a', 'b'])`:
```
irb(main):001:0> def sanitize_filter(filter_hash)
irb(main):002:1>   filter_hash.map do |key, value|
irb(main):003:2*     if value && value.is_a?(Array)
irb(main):004:3>       filtered_or = value.map { |filter| sanitize_filter(filter) }
irb(main):005:3>       [key, filtered_or]
irb(main):006:3>     else
irb(main):007:3*       [key, '?']
irb(main):008:3>     end
irb(main):009:2>   end.to_h
irb(main):010:1> end
=> :sanitize_filter
irb(main):011:0> sanitize_filter({'tags' => [1, 2]})
NoMethodError: undefined method `map' for 1:Integer
Did you mean?  tap
	from (irb):2:in `sanitize_filter'
	from (irb):4:in `block (2 levels) in sanitize_filter'
	from (irb):4:in `map'
	from (irb):4:in `block in sanitize_filter'
	from (irb):2:in `each'
	from (irb):2:in `map'
	from (irb):2:in `sanitize_filter'
	from (irb):11
```

I would change the method/algorithm to something like:
```ruby
def sanitize_filter_element(element)
  if element.is_a?(Array)
    element.map { |e| sanitize_filter_element(e) }
  elsif element.is_a?(Hash)
    element.map { |k, v| [k, sanitize_filter_element(v)] }.to_h
  else
    '?'
  end
end
```

The only concern I have about the above is that there could be giant arrays for `$in`, but I'll pull the old college textbook approach of writing "This is left as an exercise for the reader".